pkg/seachest/seachest.go

```diff
@@ -0,0 +1,116 @@
+// +build linux,cgo
+
+/*
+Copyright 2018 The OpenEBS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package seachest
+
+/*
+#cgo LDFLAGS: -lopensea-operations -lopensea-transport -lopensea-common
+#cgo CFLAGS: -I../../../openSeaChest/include -I../../../openSeaChest/opensea-common/include -I../../../openSeaChest/opensea-operations/include -I../../../openSeaChest/opensea-transport/include
+#include "common.h"
+#include "openseachest_util_options.h"
+#include "common_public.h"
+#include "ata_helper.h"
+#include "ata_helper_func.h"
+#include "scsi_helper.h"
+#include "scsi_helper_func.h"
+#include "nvme_helper.h"
+#include "nvme_helper_func.h"
+#include "cmds.h"
+#include "drive_info.h"
+#include <libudev.h>
+*/
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+// Identifier (devPath such as /dev/sda,etc) is an identifier for seachest probe
+type Identifier struct {
+	DevPath string
+}
+
+func (I *Identifier) SeachestBasicDiskInfo() (*C.driveInformationSAS_SATA, C.int) {
+
+	var device C.tDevice
+	var Drive C.driveInformationSAS_SATA
+	str := C.CString(I.DevPath)
+	defer C.free(unsafe.Pointer(str))
+
+	err := C.get_Device(str, &device)
+	err = C.get_SCSI_Drive_Information(&device, &Drive)
+	C.print_SAS_Sata_Device_Information(&Drive)
```

Comment:
should we check whether Drive is NULL or not ?

Reply:
This is struct coming from function stack. However we were not checking this against NULL in caller. I added that too. Thanks !!!